algorithms/src/fft/polynomial/dense.rs

```diff
@@ -224,10 +225,10 @@ impl<'a, F: Field> AddAssign<&'a DensePolynomial<F>> for DensePolynomial<F> {
             self.coeffs.resize(other.coeffs.len(), F::zero());
             // Zip safety: `self` and `other` have the same length.
             cfg_iter_mut!(self.coeffs).zip(&other.coeffs).for_each(|(a, b)| *a += b);
-            // If the leading coefficient ends up being zero, pop it off.
-            while self.coeffs.last().unwrap().is_zero() {
-                self.coeffs.pop();
-            }
+        }
+        // If the leading coefficient ends up being zero, pop it off.
+        while self.coeffs.last().unwrap().is_zero() {
```

Comment:
```suggestion
        while self.coeffs.last().map(|c| c.is_zero()) == Some(true) {
```
while I'm guessing it's guaranteed that there are always elements in the collection, this safe alternative is free

Reply:
febb1da7e